### PR TITLE
Compatibility with jax>=0.2.26

### PR DESCRIPTION
**Pull request by dionhaefner**
JAX moving things around again.

### CODE REVIEW

**Comment by PhilipVinc** [comment]
Would it be worth trying to maintain compatibility with older jax versions, and doing something like

```
if jax.version > 26
  Token = core.Token
else:
  Token = xla.Token
```

?

**Comment by dionhaefner** [comment]
I have no sentimentality towards older JAX versions, people can always use older mpi4jax versions if they want to stay on older JAX for whatever reasons. But if you prefer backwards compatibility we can do it.

**Comment by PhilipVinc** [comment]
I also don't really care.
But I've got users relying on old versions of jax for a variety of reasons (some compile from source jaxlib) so I'd rather not bump suddenly 1 year worth of releases if it's avoidable.

**Comment by PhilipVinc** [comment]
thanks a lot

**Comment by dionhaefner** [comment]
Let's stay compatible with Python 3.6 then while we're at it.

**Comment by PhilipVinc** [comment]
>Let's stay compatible with Python 3.6 then while we're at it.

That I care less, because Jax dropped compatibility 6 months ago https://github.com/google/jax/commit/94446ff757a3ff1c0dfaba3ecff437ae343c2c71

**Comment by dionhaefner** [comment]
I lightened the test matrix a little, no need to check everything against Python 3.6.

**Comment by dionhaefner** [comment]
Thanks! The reception has been overwhelmingly positive, so a few bad actors won't spoil that. I was glad to see you chime in with some words of reason :)

File: docs/environment.yml [hunk 1]
```diff
@@ -7,9 +7,11 @@ channels:
...
-  - jax
-  - jaxlib
   - mpi4py
   - cython
   - sphinx
   - sphinx_rtd_theme
+  - pip
+  - pip:
+    # JAX is behind on conda-forge, seems like pip is better supported
+    - jax[cpu]
```
[the remark this answers is not in the document]
Comment (dionhaefner) [reply docs/environment.yml:17]: Didn't work when I tried it today 😕

**Pull request merged**